src/App.js

```diff
@@ -0,0 +1,126 @@
+import './App.css';
+import React, {useState} from "react";
+
+import AddButton from "./addButton";
+import DeleteButton from "./deleteButton";
+import TaskList from './taskList';
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import Alert from "./Alert";
+
+import { initializeApp } from "firebase/app";
+import {
+    getFirestore,
+    query,
+    collection,
+    orderBy,
+    serverTimestamp,
+    setDoc,
+    doc,
+    deleteDoc,
+} from "firebase/firestore";
+import { useCollectionData } from "react-firebase-hooks/firestore";
+
+const firebaseConfig = {
+    apiKey: "AIzaSyAwgiUzrKROpWbF5MsvzmVZR117tTiYD9s",
+    authDomain: "cs124-lab3-23fc7.firebaseapp.com",
+    projectId: "cs124-lab3-23fc7",
+    storageBucket: "cs124-lab3-23fc7.appspot.com",
+    messagingSenderId: "850754803504",
+    appId: "1:850754803504:web:7fc3629eb6e96acfd61af5"
+};
+
+const firebaseApp = initializeApp(firebaseConfig);
+const db = getFirestore(firebaseApp);
+const collectionName = "todos";
+const priorityLevels = ["c", "b", "a"]; //the low, medium, high display is in PriorityButtons.js
+
+function App() {
+    const qCreated = query(collection(db, collectionName), orderBy("created"));
+    const [todos, loading, error] = useCollectionData(qCreated);
+    const [showAlert, setShowAlert] = useState(false);
+    const [sortByPriority, setSortByPriority] = useState(false);
+
+    const qSortPriority = query(collection(db, collectionName), orderBy("priority"));
+    const [todosPriority, loadingPriority, errorPriority] = useCollectionData(qSortPriority);
```

Comment:
Very complicated way to do this. Instead, I suggest you do a single useCollectionData based on the current state of sortByPriority. Otherwise, you have twice the network traffic (because you are obtaining all items twice).